tests/environment/customEnvironment.ts

```diff
@@ -0,0 +1,25 @@
+import Environment from "jest-environment-jsdom";
+// Requires OPENSSL_CONF=/dev/null (see https://github.com/nodejs/node/discussions/43184?sort=new) 
+import { Crypto, CryptoKey } from "@peculiar/webcrypto";
+
+// Custom test environment copied from https://github.com/jsdom/jsdom/issues/2524
+// in order to add TextEncoder to jsdom. TextEncoder is expected by jose.
+
+export default class CustomTestEnvironment extends Environment {
+  async setup() {
+    await super.setup();
+    if (typeof this.global.TextEncoder === "undefined") {
+      // The following doesn't work from jest-jsdom-polyfills.
+      // TextEncoder (global or via 'util') references a Uint8Array constructor
+      // different than the global one used by users in tests. This makes sure the
+      // same constructor is referenced by both.
+      this.global.Uint8Array = Uint8Array;
+    }
+
+    // The following can be moved to jest-jsdom-polyfills.
+    // jsdom doesn't implement the Web Crypto API
+    // For some reason,  this.global.crypto = new Crypto() leaves .subtle undefined
+    (this.global.crypto.subtle as any) = (new Crypto()).subtle;
+    this.global.CryptoKey = CryptoKey;
+  }
+};
```

Comment:
Let's fix this in jest-jsdom-polyfills instead of using custom environments.

Reply:
This works in the polyfill (I have tested it locally, I was planning on merging this and revisiting after releasing the updated polyfills), but I couldn't get the `Uint8Array` thing to work in there, so for now I'm afraid the custom environment has to stay. We can revisit why, but I'd like to move on because this has been a time sink :)

Reply:
See https://github.com/inrupt/javascript-style-configs/pull/62

Reply:
I think this should've been released now, so this shouldn't be necessary anymore.